File: core/src/nl/t64/game/rpg/screens/inventory/inventoryslot/InventorySlotsTableListener.java
package nl.t64.game.rpg.screens.inventory.inventoryslot;

import com.badlogic.gdx.Input;
import com.badlogic.gdx.scenes.scene2d.InputEvent;
import com.badlogic.gdx.scenes.scene2d.InputListener;
import lombok.AllArgsConstructor;

import java.util.function.IntConsumer;


@AllArgsConstructor
public class InventorySlotsTableListener extends InputListener {

    private final IntConsumer selectNewSlot;
    private final int slotsInRow;

    @Override
    public boolean keyDown(InputEvent event, int keycode) {
        switch (keycode) {
            case Input.Keys.UP -> selectNewSlot.accept(-slotsInRow);
            case Input.Keys.DOWN -> selectNewSlot.accept(slotsInRow);
            case Input.Keys.LEFT -> selectNewSlot.accept(-1);
            case Input.Keys.RIGHT -> selectNewSlot.accept(1);
        }
        return true;
    }

}

File: core/src/nl/t64/game/rpg/screens/inventory/inventoryslot/InventorySlotsTableListener.kt
package nl.t64.game.rpg.screens.inventory.inventoryslot

import com.badlogic.gdx.Input
import com.badlogic.gdx.scenes.scene2d.InputEvent
import com.badlogic.gdx.scenes.scene2d.InputListener


class InventorySlotsTableListener(
    private val selectNewSlot: (Int) -> Unit,
    private val slotsInRow: Int
) : InputListener() {

    override fun keyDown(event: InputEvent, keycode: Int): Boolean {
        when (keycode) {
            Input.Keys.UP -> selectNewSlot.invoke(-slotsInRow)
            Input.Keys.DOWN -> selectNewSlot.invoke(slotsInRow)
            Input.Keys.LEFT -> selectNewSlot.invoke(-1)
            Input.Keys.RIGHT -> selectNewSlot.invoke(1)
        }
        return true
    }

}
